Lib/asyncio/sslproto.py

```diff
@@ -552,6 +555,15 @@ def _start_shutdown(self):
         self._in_shutdown = True
         self._write_appdata(b'')
 
+        if self._shutdown_timeout is not None:
+            self._shutdown_timeout_handle = self._loop.call_later(
+                self._shutdown_timeout, self._start_shutdown_timeout)
+
+    def _start_shutdown_timeout(self):
+        if self._transport is not None:
```

Comment:
To make the code more clear I would set `self._transport = None` in `_finalize` and `_fatal_error` methods.